process/block/interceptedBlocks/interceptedEquivalentProof.go

```diff
@@ -0,0 +1,170 @@
+package interceptedBlocks
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	"github.com/multiversx/mx-chain-core-go/data/block"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/sharding"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+const interceptedEquivalentProofType = "intercepted equivalent proof"
+
+// ArgInterceptedEquivalentProof is the argument used in the intercepted equivalent proof constructor
+type ArgInterceptedEquivalentProof struct {
+	DataBuff          []byte
+	Marshaller        marshal.Marshalizer
+	ShardCoordinator  sharding.Coordinator
+	HeaderSigVerifier consensus.HeaderSigVerifier
+	Headers           dataRetriever.HeadersPool
```

Comment:
I don't think we need the headers pool with the latest changes in the proof structure